### PR TITLE
fix: DynamicScrapper

### PR DESCRIPTION
**Pull request by clatapie**
When merging #197, the following error appeared in the dev documentation:

image

This PR is to fix this rendering error.

### CODE REVIEW

**Comment by germa89** [comment]
Are you sure is it a cache issue? The doc run that deployed didnt use the cache.

https://github.com/ansys/pyansys-math/actions/runs/7992151629/job/21825019425#step:9:32

**Comment by clatapie** [comment]
For some reasons, when using `ansys/actions/doc-build@v5`, the DynamicScrapper doesn't render correctly.

**Comment by clatapie** [comment]
Building documentation with `ansys/action`:
image

Building documentation manually:
image

I will investigate why using `ansys/actions` prevents the `DynamicScrapper` to render correctly. Pinging @AlejandroFernandezLuces for visibility on this topic.

**Comment by clatapie** [comment]
Disabling the `check-link` option, fixes the issue with the `ansys/action`.

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -100,6 +101,7 @@ jobs:
           sudo apt-get install -y pandoc xvfb libgl1-mesa-glx
 
       - name: "Cache pip"
+        if: ${{ env.USE_CACHE == 'TRUE' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:104]:
```suggestion
        if: env.USE_CACHE == true
```

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -233,6 +235,7 @@ jobs:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
 
       - name: "Cache pip"
+        if: ${{ env.USE_CACHE == 'TRUE' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:238]:
```suggestion
        if: env.USE_CACHE == true
```

File: .github/workflows/ci_cd.yml [hunk 3]
```diff
@@ -264,6 +267,7 @@ jobs:
           echo "AnsysMath version is: $(python -c "from ansys.math.core import __version__; print(__version__)")"
 
       - name: "Cache examples"
+        if: ${{ env.USE_CACHE == 'TRUE' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:270]:
```suggestion
        if: env.USE_CACHE == true
```

File: .github/workflows/ci_cd.yml [hunk 4]
```diff
@@ -272,15 +276,16 @@ jobs:
             Examples-v${{ env.RESET_EXAMPLES_CACHE }}-${{ steps.version.outputs.ANSYSMATH_VERSION }}
 
       - name: "Cache docs build directory"
+        if: ${{ env.USE_CACHE == 'TRUE' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:279]:
```suggestion
        if: env.USE_CACHE == true
```

File: .github/workflows/ci_cd.yml [hunk 5]
```diff
@@ -272,15 +276,16 @@ jobs:
...
         uses: actions/cache@v4
         with:
           path: doc/_build
           key: doc-build-v${{ env.RESET_DOC_BUILD_CACHE }}-${{ steps.version.outputs.ANSYSMATH_VERSION }}-${{ github.sha }}
           restore-keys: |
             doc-build-v${{ env.RESET_DOC_BUILD_CACHE }}-${{ steps.version.outputs.ANSYSMATH_VERSION }}
 
-
       - name: "Cache autosummary"
+        if: ${{ env.USE_CACHE == 'TRUE' }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:288]:
```suggestion
        if: env.USE_CACHE == true
```

**Review by RobPasMue — COMMENTED** [review]
Let's use proper boolean env vars - apart from that LGTM

**Review by RobPasMue — APPROVED** [review]

**Comment by AlejandroFernandezLuces** [comment]
So, there is an issue with the interactive documentation when using the link checker from Sphinx, is that the case? Out of curiosity, when you tried manually, you were using the link checker too? 

We might have to add a regex for the links from the interactive docs to `conf.py` to ignore those links rather than not using the link checker, specially if this happens in more libraries.

**Comment by AlejandroFernandezLuces** [comment]
I'm not sure, but this adding this to the `conf.py` might work as well, feel free to try it if you have the time 🙂 

``` python
linkcheck_ignore = [r'*.vtksz']

```

**Comment by RobPasMue** [comment]
> So, there is an issue with the interactive documentation when using the link checker from Sphinx, is that the case? Out of curiosity, when you tried manually, you were using the link checker too?
> 
> We might have to add a regex for the links from the interactive docs to `conf.py` to ignore those links rather than not using the link checker, specially if this happens in more libraries.

I agree with @AlejandroFernandezLuces - let's ignore the files where this happens, or the extensions to the files involved.

**Pull request merged**